src/System.Net.WebSockets.WebSocketProtocol/tests/WebSocketProtocolTests.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.WebSockets.Tests
+{
+    public sealed class WebSocketProtocolTests
+    {
+        [Fact]
+        public void CreateFromStream_InvalidArguments_Throws()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("stream",
+                () => WebSocketProtocol.CreateFromStream(null, true, "subProtocol", TimeSpan.FromSeconds(30), default(Memory<byte>)));
+            AssertExtensions.Throws<ArgumentException>("stream",
+                () => WebSocketProtocol.CreateFromStream(new MemoryStream(new byte[100], writable:false), true, "subProtocol", TimeSpan.FromSeconds(30), default(Memory<byte>)));
+            AssertExtensions.Throws<ArgumentException>("stream",
+                () => WebSocketProtocol.CreateFromStream(new UnreadableStream(), true, "subProtocol", TimeSpan.FromSeconds(30), default(Memory<byte>)));
+
+            AssertExtensions.Throws<ArgumentException>("subProtocol",
+                () => WebSocketProtocol.CreateFromStream(new MemoryStream(), true, "    ", TimeSpan.FromSeconds(30), default(Memory<byte>)));
+            AssertExtensions.Throws<ArgumentException>("subProtocol",
+                () => WebSocketProtocol.CreateFromStream(new MemoryStream(), true, "\xFF", TimeSpan.FromSeconds(30), default(Memory<byte>)));
+
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("keepAliveInterval", () =>
+                WebSocketProtocol.CreateFromStream(new MemoryStream(), true, "subProtocol", TimeSpan.FromSeconds(-2), default(Memory<byte>)));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(14)]
+        [InlineData(4096)]
+        public void CreateFromStream_ValidBufferSizes_Succeed(int bufferSize)
+        {
+            Assert.NotNull(WebSocketProtocol.CreateFromStream(new MemoryStream(), false, null, Timeout.InfiniteTimeSpan, new Memory<byte>(new byte[bufferSize])));
+            Assert.NotNull(WebSocketProtocol.CreateFromStream(new MemoryStream(), true, null, Timeout.InfiniteTimeSpan, new Memory<byte>(new byte[bufferSize])));
+        }
+
+        [OuterLoop] // Connects to external server.
+        [Fact]
+        [ActiveIssue(26288)]
+        public async Task WebSocketProtocol_CreateFromConnectedStream_Succeeds()
+        {
+            Socket s = new Socket(SocketType.Stream, ProtocolType.Tcp);
+            await s.ConnectAsync("corefx-net.cloudapp.net", 80);
+            using (NetworkStream ns = new NetworkStream(s))
+            {
+                using (WebSocket socket = WebSocketProtocol.CreateFromStream(ns, false, null, TimeSpan.FromSeconds(10)))
```

Comment:
This is buggy.  As mentioned in https://github.com/dotnet/corefx/pull/26296/files#r161290252, CreateFromStream is meant to be used with a stream that's already connected for websocket communication; this isn't.

Reply:
This also isn't connecting to an echo WebSocket, so even if the connection was upgraded, the receive would still hang as there's nothing to receive.

Reply:
Yeah changing the test.